config/cmake/addlibzmq.cmake

```diff
@@ -159,11 +159,21 @@ endif()
 if(HELICS_BUILD_CXX_SHARED_LIB OR NOT HELICS_DISABLE_C_SHARED_LIB)
 
 	if(NOT HELICS_USE_ZMQ_STATIC_LIBRARY)
-		install(
-			FILES $<TARGET_FILE:${zmq_target_output}>
-			DESTINATION ${CMAKE_INSTALL_BINDIR}
-			COMPONENT libs
-		)
+		set_target_properties(${zmq_target_output} PROPERTIES PUBLIC_HEADER "")
+		if(NOT ${CMAKE_VERSION} VERSION_LESS "3.13")
```

Comment:
Is this going to work properly with a package builder on windows/linux?    Just wondering if the WIN32 clause maybe should be first before the version 3.13 check of CMAKE.

Reply:
CMake does the right thing for Windows and places .dll files (zmq, libhelicsShared, etc) in the bin folder, and .lib files in the lib folder.

Reply:
The way some of the HELICS libraries are getting installed right now with an install TARGET_FILE command might be redundant, but that's another issue.

Reply:
Can we verify that this generates the correct package with ZMQ on windows with CMAKE>=3.13.

Reply:
Should we just disable packaging completely on windows if cmake<3.13 and ZMQ non-static subproject is used?

Reply:
Yes, the default CMake version included in the Windows GitHub Actions images has been >3.13 for a while now. The shared library archives from the most recent build are at https://github.com/nightlark/HELICS/releases/tag/vLinuxFix.17

I noticed the helics app binaries weren't getting included, since the executables getting built depend on the shared library being enabled. It looks like there is no way to create an installer with the helics apps that doesn't also include the helics app shared library. I'll run another test build for the installers to confirm that's working again in a bit.

Reply:
I think users building with ZMQ as a static subproject would make their lives easier, so they never need to worry about forgetting to copy the ZMQ dll to the right place. It would also simplify this part of the addlibzmq.cmake file.

Reply:
Lets review this over the next couple weeks make sure we have the right defaults and error conditions, and see if we can resolve the lingering questions of what we can and cannot do with ZMQ licensing to make sure any packaged distributions are BSD compliant.

Reply:
Alright. For now I added a warning about an unsupported configuration for installing from source if a Linux or macOS build with CMake <3.13 is using ZeroMQ as a subproject.

Reply:
The final test build is at https://github.com/nightlark/HELICS/releases/tag/vLinuxFix.18
From going through the different tar and zip files, it looks like the expected files are in each one.